### PR TITLE
Add test-adapter build and stale vendor directory verification to CI

### PR DESCRIPTION
**Pull request by dgrisonnet**
Add CI jobs verifying that the test-adapter build and the vendor directory doesn't go stale.

/assign @s-urbaniak

### CODE REVIEW

**Comment by dgrisonnet** [comment]
@johanneswuerbach can please take a look at this?

I fixed the various build issues I faced and added some CI checks to prevent such problems from slipping under our radar next time.

**Comment by johanneswuerbach** [comment]
lgtm 👍

File: .travis.yml [hunk 1]
```diff
@@ -3,8 +3,13 @@ language: go
...
-    - GO111MODULE="on"
+jobs:
+  include:
+    - stage: all
+      name: Verify
+      script: make verify && git diff --exit-code
+    - name: Build test adapter
+      script: make build-test-adapter
+    - name: Unit tests
+      script: make test
```
Comment (s-urbaniak) [inline .travis.yml:14]: `make verify` also executes tests is it is dependent on the `test` target in the makefile.

File: Makefile [hunk 2]
```diff
@@ -35,21 +29,26 @@ verify-gofmt:
 gofmt:
 	./hack/gofmt-all.sh
 
-verify: verify-gofmt test
+verify: vendor verify-gofmt
```
Comment (s-urbaniak) [inline Makefile:32]: ah, nevermind, i see you removed it.

**Pull request merged**